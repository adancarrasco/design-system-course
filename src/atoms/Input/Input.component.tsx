import * as React from 'react'
import styled from '@emotion/styled'

export interface IInputProps extends IInputStyledProps {
  placeholder?: string
  value?: string
  onChange?: () => void
  onBlur?: () => void
}

const Input: React.FC<IInputProps> = ({ placeholder, value, onChange, onBlur }) => (
  <InputStyled
    autoFocus
    placeholder={placeholder}
    value={value}
    onChange={onChange}
    onBlur={onBlur}
  />
)

interface IInputStyledProps {
  borderColor?: string
}

const InputStyled = styled.textarea<IInputStyledProps>`
  border: 1px solid;
  border-radius: 4px;
  border-color: ${(props) => props.borderColor ?? '#6868ff'};
  box-sizing: border-box;
  font-size: 0.9rem;
  padding: 0.5rem;
  resize: none;
  width: 100%;
  min-height: 20px;
`

export default Input
