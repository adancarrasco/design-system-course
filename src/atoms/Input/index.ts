export { default } from './Input.component'
export type { IInputProps } from './Input.component'
