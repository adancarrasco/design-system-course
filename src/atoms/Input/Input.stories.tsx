import React from 'react'
import { storiesOf } from '@storybook/react'

import Input from '.'

storiesOf('Atoms|Input', module)
  .add('Default', () => (
    <div style={{ width: '500px' }}>
      <Input placeholder="Add more detailed description..." />
    </div>
  ))
  .add('With value', () => (
    <div style={{ width: '500px' }}>
      <Input value="This is a super important description" />
    </div>
  ))
