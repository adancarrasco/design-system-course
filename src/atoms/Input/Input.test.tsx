import React from 'react'
import { render } from '@testing-library/react'
import { matchers } from 'jest-emotion'
import '@testing-library/jest-dom/extend-expect'

import Input, { IInputProps } from '.'

expect.extend(matchers)

describe('Input Component', () => {
  const renderComponent = (props: IInputProps, dataTestId: string) =>
    render(<Input data-testid={dataTestId} {...props} />)

  it('matches snapshot and default render', () => {
    const dataTestId = 'input'
    const props = {
      placeholder: 'Add more detailed description...',
      value: 'This is the input value',
      onChange: jest.fn(),
    }
    const { getByText } = renderComponent(props, dataTestId)
    const container = getByText('This is the input value')
    expect(container).toMatchSnapshot()
    expect(container).toBeInTheDocument()
  })
})
