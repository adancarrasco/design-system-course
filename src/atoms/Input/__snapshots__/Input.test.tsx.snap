// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches snapshot and default render 1`] = `
.emotion-0 {
  border: 1px solid;
  border-radius: 4px;
  border-color: #6868ff;
  box-sizing: border-box;
  font-size: 0.9rem;
  padding: 0.5rem;
  resize: none;
  width: 100%;
  min-height: 20px;
}

<textarea
  class="emotion-0"
  placeholder="Add more detailed description..."
>
  This is the input value
</textarea>
`;
